AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine+DeleteTransaction.swift

```diff
@@ -11,9 +11,43 @@ import Foundation
 import AWSPluginsCore
 
 extension StorageEngine {
+
+    enum DeleteInput {
+        case withId(id: Model.Identifier, predicate: QueryPredicate?)
+        case withPredicate(predicate: QueryPredicate)
```

Comment:
would having three separates cases `withId(id:)`, `withIdAndPredicate(id:predicate:)`, `withPredicate(predicate:)` and a static factory method that returns the proper case simplify the `predicate` property and make the `isDeleteByIdWithPredicate` property unnecessary?

Reply:
i tried this out and it solves the original problem much better! thanks, i've updated to the 3 cases and removed `isDeleteByIdWithPredicate` since i can just check the case `withIdAndPredicate`

Reply:
thanks for the change!